cdap-api/src/main/java/co/cask/cdap/api/common/RuntimeArguments.java

```diff
@@ -42,7 +40,11 @@ private RuntimeArguments() {
   public static Map<String, String> fromPosixArray(String[] args) {
     Map<String, String> kvMap = new HashMap<>();
     for (String arg : args) {
-      kvMap.putAll(Splitter.on("--").omitEmptyStrings().trimResults().withKeyValueSeparator("=").split(arg));
+      int idx = arg.indexOf('=');
+      int keyOff = arg.startsWith("--") ? "--".length() : 0;
+      String key = idx < 0 ? arg.substring(keyOff) : arg.substring(keyOff, idx);
+      String value = idx < 0 ? "" : arg.substring(idx + 1);
```

Comment:
Actually it won't. The substring requires the begin index > 0 and <= string.length.